docs/developer-guide.md

```diff
@@ -0,0 +1,210 @@
+- [ ] Publish website
+
+
+## Publish
+
+### Production Release
+
+1. Log into an authorized [npmjs.com](https://www.npmjs.com/) account. You can use [npmrc](https://www.npmjs.com/package/npmrc) to manage multiple npm profiles.
+2. Make sure both the `master` and the release branch are up to daate:
+
+    ```bash
+    $ git checkout master
+    $ git pull
+    $ git checkout 3.3-release
+    $ git pull
+    ```
+
+3. Find the last release commit on the `<latest>-release` branch:
+
+    ```bash
+    $ git log
+
+    ...
+
+    commit a05d23059444cf29bb4e38ea5e4cd6172a4f463d
+    Author: Xiaoji Chen <xiaoji@uber.com>
+    Date:   Fri Aug 3 10:11:45 2018 -0700
+
+        3.3.3
+    ```
+
+    Find the commits on `master` after this commit:
+
+    ```bash
+    $ git log master --since="`git show -s --format=%ci a05d2305`"
+    ```
+
+4. Find out which new commits should be cherry-picked into the release branch.
+
+    If a commit is a bug fix for the current production release, or a minor, non-breaking new feature, it can be published as a patch. When in doubt, check the original PR's milestone label on GitHub. For example [#565](https://github.com/uber/react-map-gl/pull/565):
+
+    ```bash
+    $ git cherry-pick 1238140a
+    ```
+
+    And add its commit message to CHANGELOG.md:
+
+    ```
+    ## 3.3.4 (Aug 4, 2018)
+
+    - fix capture* props for overlay components (#565)
+    ```
+
+5. Commit the changelog:
+
+    ```bash
+    $ git add .
+    $ git commit -m "3.3.4 Changelog"
+    ```
+
+6. Bump version and publish:
+
+    ```bash
+    $ npm version patch
+    $ npm run publish-prod
+    ```
+
+7. If the new patch release fixes a bug on the website, republish the website.
+
+
+### Beta Release
+
+1. Log into an authorized [npmjs.com](https://www.npmjs.com/) account. You can use [npmrc](https://www.npmjs.com/package/npmrc) to manage multiple npm profiles.
+2. Make sure the `master` branch is up to daate:
+
+    ```bash
+    $ git checkout master
+    $ git pull
+    ```
+
+3. Find the last release commit on the `master` branch:
+
+    ```bash
+    $ git log
+
+    ...
+    commit a05d23059444cf29bb4e38ea5e4cd6172a4f463d
+    Author: Xiaoji Chen <xiaoji@uber.com>
+    Date:   Fri Aug 3 10:11:45 2018 -0700
+
+        4.0.0-alpha.1
+    ```
+
+    Find the commits on `master` after this commit:
+
+    ```bash
+    $ git log --since="`git show -s --format=%ci a05d2305`"
+    ```
+
+4. And add new commit messages to CHANGELOG.md.
+
+5. Manually bump the verion in `package.json` to the next pre-release version.
+
+6. Commit and publish:
+
+    ```bash
+    $ git add .
+    $ git commit -m "4.0.0-alpha.2"
+    $ npm run publish-beta
+    $ git push
+    ```
+
+### Website
+
+1. Make sure the `<latest>-release` branch is up to daate:
+
+    ```bash
+    $ git checkout 3.3-release
+    $ git pull
+    ```
+
+2. Test the website:
+
+    ```bash
+    $ cd website
+    $ yarn
+    $ yarn start
+    ```
+
+3. Build and publish the website:
```

Comment:
Because of npm permissions, no one outside the core team can publish.

Should we separate into instructions for external contributors and core team?